lisp/transient.el

```diff
@@ -512,6 +512,17 @@ slot is non-nil."
    (argument-regexp  :initarg :argument-regexp))
   "Class used for sets of mutually exclusive command-line switches.")
 
+(cl-defmethod initialize-instance ((obj transient-switches) slots)
+  (cl-call-next-method)
+  (-let [(&plist :argument-regexp :choices :argument-format) slots]
+    (unless argument-regexp
+      (setf (oref obj argument-regexp)
+            (--> choices
+                 (-map #'regexp-quote it)
+                 (regexp-opt it t)
```

Comment:
`regexp-opt` takes plain strings, so the `regexp-quote` before it is not correct.

Reply:
Fixed, Plz check. @npostavs

Reply:
@cireu I think it's okay, though I'll let others make a full review since I'm unfamiliar with the CLOS stuff in general, and the transient classes in particular.